Classes/Service/DemandService.php

```diff
@@ -322,6 +323,41 @@ protected function getPropertiesForDemandedTables(array $tables, array $demands)
             }
         }
 
-        return array_merge_recursive($demandedProperties, $demands);
+        $filteredDemands = $this->filterDemandedProperties($demandedProperties, $demands);
+
+        return array_merge_recursive($demandedProperties, $filteredDemands);
+    }
+
+    /**
+     * @param QueryBuilder $queryBuilder
+     */
+    public function getSortBy(QueryBuilder $queryBuilder): void
+    {
+        $demands = $this->getDemandsFromDemandProviders();
+        $queryBuilder->orderBy($demands['orderBy'], $demands['orderDirection']);
+    }
```

Comment:
I'm missing a `$tables` argument here. What if the table in `$demands['orderBy']` is not in the query?